tests/test_typeddict_factory.py

```diff
@@ -44,3 +45,23 @@ class MyFactory(ModelFactory[MyModel]):
     assert result.td["name"]
     assert result.td["list_field"][0]
     assert type(result.td["int_field"]) in (type(None), int)
+
+
+def test_typeddict_with_required_and_non_required_fields() -> None:
+    class TypedDictModel(TypedDict):
+        id: Required[int]
+        name: NotRequired[str]
```

Comment:
No assertion for `name` below, is that by design?

Reply:
Nope. I just missed checking for `name`.